dali/test/dali_operator_test.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_TEST_DALI_OPERATOR_TEST_H_
+#define DALI_TEST_DALI_OPERATOR_TEST_H_
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <dali/pipeline/pipeline.h>
+#include "operators_graph.h"
+#include "operator_arguments.h"
+
+namespace dali {
+
+namespace testing {
+
+template<typename InputBackend, typename OutputBackend>
+class DaliOperatorTest : public ::testing::Test {
+
+ public:
+  template<typename Backend>
+  using DataType = std::unique_ptr<dali::TensorList<Backend>>;
+
+  /**
+   * Verify(output, anticipated_output)
+   */
+  using Verify = std::function<void(const dali::TensorList<OutputBackend> &,
```

Comment:
What about heterogeneous outputs?

Reply:
`std::vector` for instance, but we don't need them right now, since it's not likely I'll implement them in next PR. I want to to this gradually. That's why this PR is just an API, which will be extended

Reply:
`std::vector` can't handle heterogeneous tensors (i.e. different backends). Having a function that handles each output separately is appealing, but it's hard to verify outputs of different type (or otherwise have different validation methods for different outputs).

Reply:
If we want to have such assumption we need to completely rework API of this test.
So:
![image](https://user-images.githubusercontent.com/39967756/49739950-6209c680-fc93-11e8-941f-4e6d410a9269.png)
DaliOperatorTesth that has only SetInput which accepts CPU tensorlists and keep that one - for input and reference oputput.
RunTest could still accept operator_arguments and replace values provided in the graph defined as a set of OpSpecs. 
GenerateOperatorsGraph should accept operator_arguments and return set of configured OpSpecs. RunTest should call GenerateOperatorsGraph to get execution pipeline.
Then some arguments could be hard coded and some set in the runtime.
Verify should accept DeviceWorkspace returned by RunTest and it should extract outputs, copy back to the CPU and compare with the reference data from SetInput.

Reply:
One of the assumptions of new pipeline was that it will be independent from existing Dali routines. The rationale was that in case anything in Dali changes (e.g. `OpSpec`, `Workspace` - and we know for sure they will change), all the tests have to be reimplemented. This pipeline isn't intended to be used for regression test (for unit testing either), but rather a verification test for given graph of operators, i.e. answering a question

> Does graph of operators `G`, for input `In` and arguments `Args` provide output `Out`?